<?xml version="1.0"?>
<launch>

<arg name="model" default="$(find diff_drive)/urdf/02-diff_drive.urdf.xacro"/>
<arg name="rvizconfig" default="$(find urdf_tutorial)/rviz/urdf.rviz"/>


<!-- Launch gazebo -->
<include file="$(find diff_drive)/launch/gazebo.launch">
  <arg name="model" value="$(arg model)"/>
 </include>

<node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)"/>

<!-- Load joint controller configuration from YAML file to parameter server -->

<rosparam file="$(find diff_drive)/config/joint_states.yaml" command="load"/>

<rosparam file="$(find diff_drive)/config/diff_drive.yaml" command="load"/>


<!-- load the controllers -->
<node name="robot_arm_controller_spawner" pkg="controller_manager" type="spawner"
  respawn="true" output="screen"
   args="/Diff_Drive/joint_state_controller
         /Diff_Drive/diff_drive_controller
        --shutdown-timeout 3"/>

<!-- convert joint states to TF transforms for rviz, etc -->
<node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
    respawn="false" output="screen">
</node>

<node name="rqt_robot_steering" pkg="rqt_robot_steering" type="rqt_robot_steering">
  <param name="default_topic" value="/Diff_Drive/diff_drive_controller/cmd_vel"/>
</node>


<!-- multi laser scan merge -->
<node pkg="ira_laser_tools" name="laserscan_multi_merger" type="laserscan_multi_merger" output="screen">
        	<param name="destination_frame" value="chassis"/>
		<param name="cloud_destination_topic" value="/merged_cloud"/>
		<param name="scan_destination_topic" value="/scan_multi"/>
                <param name="laserscan_topics" value ="/Diff_Drive/laser/scan1 /Diff_Drive/laser/scan2" /> <!-- LIST OF THE LASER SCAN TOPICS TO SUBSCRIBE -->

                <param name="angle_min" value="-3.141592"/>
                <param name="angle_max" value="3.141592"/>
                <param name="angle_increment" value="0.0058"/>
                <param name="scan_time" value="0.0333333"/>
                <param name="range_min" value="0.30"/>
                <param name="range_max" value="50.0"/>

	</node>

</launch>
